src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs

```diff
@@ -2132,6 +2124,42 @@ private ReadOnlyCollection<TValue> GetValues()
             }
         }
 
+        /// <summary>Whether T's type can be written atomically (i.e., with no danger of torn reads).</summary>
+        private static bool IsWriteAtomic<T>()
```

Comment:
Have you checked the code for diffs for this? I suspect that the method won't get inlined and there is going to be a call (regression from current main).